### PR TITLE
Updates for `esbuild` tree shaking

### LINKED ISSUE #237: Investigate allowing esbuild to tree shake library
I discovered that esbuild does not tree shake the library at all (https://github.com/evanw/esbuild/issues/1573). The issue might be in the way each of the classes assigns `class` and `prototype` for the factory function. Need to investigate changing this.

```js
export default function factory() {
  return new Vector(...arguments);
}
factory.prototype = Vector.prototype;
factory.class = Vector;
```

### PR DESCRIPTION
**Pull request by anderoonies**
fixes #237 

## Overview
Removes the use of `.prototype` on factory functions, instead exporting classes directly, which allows `esbuild` to ignore side effects that might occur in the `.prototype` assignment

Update documentation to reflect the import and use of classes.

## Testing `esbuild` tree shaking
I put together a test game to test `esbuild` and `rollout` tree shaking:
```js
// game.js
import { init, Sprite, GameLoop } from "../kontra.mjs";

init();

let sprite = Sprite({
  x: 100,
  y: 100,
  dx: 2,
  width: 20,
  height: 40,
  color: "red",
});

let loop = GameLoop({
  update() {
    sprite.update();
  },
  render() {
    sprite.render();
  },
});

loop.start();
```

The commands for generating the bundles:
```
npx esbuild game.js --bundle --format=iife --outfile=esbuild.game.js
npx rollup game.js --format iife --file rollup.game.js 
```

The output for each:
|   |  # lines | size (KB)  |
|---|---|---|
|  `esbuild` | 493  |  11 KB |
| `rollout`  |  1667 | 46K  | 

## Code formatting
I also made a pass on the `src/` and `test/` directories with the Prettier code formatter, since my IDE had automatically formatted things when working on this. I used the settings:
```json
{
  "arrowParens": "avoid",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 100,
  "proseWrap": "preserve",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "none",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "parser": "babel"
}
```
which are pretty near default, and might be a good starting point if you want to start formatting the entire project. I added `"arrowParens": "avoid"`, `"singleQuote": true`, `"tabWidth": 2` and `"trailingComma": "none"` to try to match the existing style.
I just put that json in a `.prettierrc` and ran `prettier --write src/**/*.js` and `prettier --write test/**/*.js` with Prettier 2.2.1.

### CODE REVIEW

**Review by straker — DISMISSED** [review]
Looks good! So nice to have prettier on, though I will miss my 5 newlines in tests :)

Just one thing I noticed but I think everything else looks good, though I'm curious why rollup's output is bigger / different than esbuilds. I'll try it out and see if I notice anything.

**Comment by anderoonies** [comment]
Alright @straker I've included the `.prettierrc.js` file and additionally run `npx prettier --write **/*.js` on the project now that you're sure about that config.

`kontra.defaults.js` also includes the `Class`es, let me know if there's anything else you'd need from me!

File: package-lock.json [hunk 1]
```diff
@@ -1,6568 +0,0 @@
-{
```
Comment (straker) [inline package-lock.json:1]: What was the reasoning of deleting this file?

**Comment by anderoonies** [comment]
Hm I just intended to restore it to what it was on master. I’ve been seeing
package-lock.json diffs not working all day today, it didn’t show that a
package-lock was in the project at all prior to this PR and I didn’t want
to add one if you weren’t already keeping one in version control.
I’ll generate a new one including prettier and include it.

**Comment by straker** [comment]
npm v7 has a few major bugs in it that will prevent the tests from working. The main bug is that it [resolves https github links as ssh](https://github.com/npm/cli/issues/2610), requiring ssh authentication. You'll need to downgrade to npm v6 to get the tests working if you're going to generate a new package-lock file.

**Comment by anderoonies** [comment]
Will do. I can leave the package-lock out of my changes, but will probably want one with prettier included.

**Comment by anderoonies** [comment]
`package0lock.json` is updated, thanks for the information about npmv7 as well. haven't seen that come up.

**Comment by straker** [comment]
Thanks for putting this all together. It'll be nice to have esbuild be able to tree shake the library.

**Pull request merged**